### PR TITLE
support attach metadata in PullImage CRI interface

### LINKED ISSUE #1155: [feature request] Support attach metadata in PullImage CRI interface during ImagePulljobs
**What would you like to be added**:

When kubelet creates pods, kubelet will attach pod metadata as podSandboxConfig params in the PullImage CRI interface, however if we download image using ImagePulljobs, currently there is no way to attach user metadata into CRI interface, which makes some function infeasible when using ImagePulljobs. 

```
apiVersion: apps.kruise.io/v1alpha1
kind: ImagePullJob
metadata:
  name: job-with-annotations
spec:
  image: nginx:1.9.1  
  # add new fields to specifiy sandbox config
  SandboxConfig:
      annotations:
          io.kubernetes.image.metrics.tags: "app=foo"           
```

**Why is this needed**:
 Image registry can support tag image download traffic with custom label, so that image registry can identify and limit large traffic.

### PR DESCRIPTION
**Pull request by diannaowa**
### Ⅰ. Describe what this PR does
Support attach metadata in PullImage CRI interface during ImagePulljobs;

```
apiVersion: apps.kruise.io/v1alpha1
kind: ImagePullJob
metadata:
  name: job-with-always
spec:
  image: nginx
  sandboxConfig:
    annotations:
      io.kubernetes.image.metrics.tags: "app=foo"
    labels:
      app: foo
  parallelism: 10
```

NodeImage:

```
apiVersion: apps.kruise.io/v1alpha1
kind: NodeImage
metadata:
  creationTimestamp: "2023-02-24T17:26:55Z"
  generation: 4
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    env: test-containerd
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: vm-0-10-ubuntu
    kubernetes.io/os: linux
  name: vm-0-10-ubuntu
  resourceVersion: "23916206"
  uid: 8416e50e-7dd0-4cce-8d13-9425c3afd020
spec:
  images:
    nginx:
      sandboxConfig:
        annotations:
          io.kubernetes.image.metrics.tags: "app=foo"
        labels:
          app: foo
      tags:
      - createdAt: "2023-02-24T17:44:03Z"
        ownerReferences:
        - apiVersion: apps.kruise.io/v1alpha1
          kind: ImagePullJob
          name: job-with-always
          namespace: default
          uid: 64f5b96a-8596-4688-8e15-c1e7adddd209
        pullPolicy:
          backoffLimit: 3
          timeoutSeconds: 600
          ttlSecondsAfterFinished: 2246
        tag: latest
       ......
```

### Ⅱ. Does this pull request fix one issue?

fixes #1155

### CODE REVIEW

File: pkg/daemon/criruntime/imageruntime/containerd.go [hunk 1]
```diff
@@ -21,6 +21,7 @@
 	"crypto/tls"
 	"encoding/json"
 	"fmt"
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
```
[the remark this answers is not in the document]
Comment (diannaowa) [reply pkg/daemon/criruntime/imageruntime/containerd.go]: done

**Review by furykerry — COMMENTED** [review]
/lgtm

**Comment by furykerry** [comment]
@diannaowa plz squash multiple commits into one

**Comment by diannaowa** [comment]
> @diannaowa plz squash multiple commits into one

Will do

**Comment by diannaowa** [comment]
Squash multiple commits into one.
Please review it. @furykerry

**Comment by diannaowa** [comment]
@zmberg

**Comment by diannaowa** [comment]
> @diannaowa Current containerd.go's PullImage function can't meet this scenarios, and i think we can use cri.go in containerd. So i suggest as follows in [factory.go](https://github.com/openkruise/kruise/blob/master/pkg/daemon/criruntime/factory.go) :
> 
> ```
> switch cfg.runtimeType {
> case ContainerRuntimeDocker:
> 	imageService, err = runtimeimage.NewDockerImageService(cfg.runtimeURI, accountManager)
> 	if err != nil {
> 		klog.Warningf("Failed to new image service for %v (%s, %s): %v", cfg.runtimeType, cfg.runtimeURI, cfg.runtimeRemoteURI, err)
> 		continue
> 	}
> case ContainerRuntimePouch:
> 	imageService, err = runtimeimage.NewPouchImageService(cfg.runtimeURI, accountManager)
> 	if err != nil {
> 		klog.Warningf("Failed to new image service for %v (%s, %s): %v", cfg.runtimeType, cfg.runtimeURI, cfg.runtimeRemoteURI, err)
> 		continue
> 	}
> case ContainerRuntimeContainerd, ContainerRuntimeCommonCRI:
> 	addr, _, err := kubeletutil.GetAddressAndDialer(cfg.runtimeRemoteURI)
> 	if err != nil {
> 		klog.Warningf("Failed to get address for %v (%s, %s): %v", cfg.runtimeType, cfg.runtimeURI, cfg.runtimeRemoteURI, err)
> 		continue
> 	}
> 	imageService, err = runtimeimage.NewCRIImageService(addr, accountManager)
> 	if err != nil {
> 		klog.Warningf("Failed to new image service for %v (%s, %s): %v", cfg.runtimeType, cfg.runtimeURI, cfg.runtimeRemoteURI, err)
> 		continue
> 	}
> }
> ```

Yeah.That is a good idea. My thought was that after this feature was done, we would look at optimizing commonCRI and containerd, since much same behavior in during pullimge with common CRI and containerd.
If we need do this, I immediately follow this code to commit a new commit.

**Comment by diannaowa** [comment]
> > If we need do this, I immediately follow this code to commit a new commit. And then the Progress of pull image will be not work when runtime is Containerd.
> 
> Yes, you can do this.

Another commit in this PR. Please review.

**Pull request merged**